internal/uid/uid.go

```diff
@@ -0,0 +1,16 @@
+package uid
+
+import (
+	"os"
+	"strconv"
+)
+
+// GetRootlessUID returns the UID of the user in the parent userNS
```

Comment:
```suggestion
// GetRootlessUID returns the UID of the current user (in the parent userNS, if any)
//
// Podman and similar software, in “rootless” configuration, when run as a non-root
// user, very early switches to a user namespace, where Geteuid() == 0 (but does not
// switch to a limited mount namespace); so, code relying on Geteuid() would use 
// system-wide paths in e.g. /var, when the user is actually not privileged to write to
// them, and expects state to be stored in the home directory.
//
// If Podman is setting up such a user namespace, it records the original UID in an
// environment variable, allowing us to make choices based on the actual user’s identity.
```
or something like that (if the above is accurate), to give newcomers more context.